gitensite/apps/bookinfo/models.py

```diff
@@ -42,8 +37,25 @@ class Book(models.Model):
     full_text = models.TextField(default="", null=True, blank=True)
     num_downloads = models.IntegerField(default=0)
     
+    added = models.DateTimeField(auto_now_add=True, null=True)
+    updated = models.DateTimeField(auto_now_add=True, null=True)
+    
     yaml = models.TextField(null=True, default="")
 
+    # using a custom save method in order to update the "updated" timestamp when specific fields are updated
```

Comment:
You don't need this method, and a custom save method is almost always a bad idea.
1. Instead, use https://docs.djangoproject.com/en/2.0/ref/models/fields/#django.db.models.DateField.auto_now
2. if you need to do something else on save, use the post_save or pre_save signals https://docs.djangoproject.com/en/2.0/ref/signals/#post-save

Reply:
3. Also, use the https://docs.djangoproject.com/en/2.0/ref/models/fields/#django.db.models.DateField.auto_now_add for the created field.

Reply:
Regarding 1 and 3, we are using this features already.

For 2, we did some research, and apparently the pre_save and post_save signals do not provide any way to determine _which_ fields are being modified. We need this because we don't want to the updated timestamp to change when the download counter is incremented. The custom save method seems to be the only reasonable way to accomplish this.

Reply:
looks like you're right, https://stackoverflow.com/questions/1355150/django-when-saving-how-can-you-check-if-a-field-has-changed/28094442#28094442